include/functional/detail/concepts.hpp

```diff
@@ -8,14 +8,14 @@
 
 #include <expected>
 #include <optional>
-#include <tuple>
 
 namespace fn::detail {
-
 template <typename T> constexpr bool is_some_expected = false;
 template <typename T, typename Err>
 constexpr bool is_some_expected<std::expected<T, Err>> = true;
 template <typename T, typename Err>
+constexpr bool is_some_expected<std::expected<T, Err> const> = true;
```

Comment:
Should we also consider `const volatile` and all of the other variations to cover all cases?

Reply:
`volatile` is deprecated, if someone keeps using it then their use cases is probably outside of the scope of this library.

Reply:
volatile is not deprecated, certain operations on volatile were deprecated. That said, volatile shouldn't be used for any of these types, you're not doing I/O with them.

Reply:
I agree none of this should be volatile. It was more a question of completeness. Technically it's still "some" 'std::expected' even if making one does not really make sense 😅

Reply:
The trade-off is 2x compile-time increase. Partial template specializations are checked linearly, and then best-match determined.